prov/bgq/Makefile.include

```diff
@@ -0,0 +1,78 @@
+if HAVE_BGQ
+AM_CFLAGS += -Wno-ignored-qualifiers
+AM_CPPFLAGS += -I$(builddir)/prov/bgq/include -I$(srcdir)/prov/bgq/include
+_bgq_files = prov/bgq/src/fi_bgq_atomic.c \
+		prov/bgq/src/fi_bgq_av.c \
+		prov/bgq/src/fi_bgq_spi.c \
+		prov/bgq/src/fi_bgq_cm.c \
+		prov/bgq/src/fi_bgq_cntr.c \
+		prov/bgq/src/fi_bgq_cq.c \
+		prov/bgq/src/fi_bgq_domain.c \
+		prov/bgq/src/fi_bgq_ep.c \
+		prov/bgq/src/fi_bgq_fabric.c \
+		prov/bgq/src/fi_bgq_info.c \
+		prov/bgq/src/fi_bgq_init.c \
+		prov/bgq/src/fi_bgq_mr.c \
+		prov/bgq/src/fi_bgq_msg.c \
+		prov/bgq/src/fi_bgq_rma.c \
+		prov/bgq/src/fi_bgq_sep.c \
+		prov/bgq/src/fi_bgq_tagged.c \
+		prov/bgq/src/fi_bgq_node.c \
+		prov/bgq/src/fi_bgq_progress.c \
+		prov/bgq/src/fi_bgq_pmi.c \
+		prov/bgq/external/memory_impl.c
+bgq_CPPFLAGS += -I@bgq_driver@ -I@bgq_driver@/spi/include/kernel/cnk
+bgq_LDFLAGS += -static
+
+prov/bgq/external/memory_impl.c: @bgq_external_source@/spi/src/kernel/cnk/memory_impl.c
+	mkdir -p $(@D)
+	cp $< $@
+
+if HAVE_BGQ_DL
+pkglib_LTLIBRARIES += libbgq-fi.la
+libbgq_fi_la_SOURCES = $(_bgq_files) $(common_srcs)
+libbgq_fi_la_CPPFLAGS = $(AM_CPPFLAGS) $(bgq_CPPFLAGS)
+libbgq_fi_la_LDFLAGS = \
+    -module -avoid-version -export-dynamic $(bgq_LDFLAGS)
+libbgq_fi_la_LIBADD = $(linkback) $(bgq_LIBS)
+libbgq_fi_la_DEPENDENCIES = $(linkback)
+else
+src_libfabric_la_SOURCES += $(_bgq_files)
+src_libfabric_la_CPPFLAGS += $(bgq_CPPFLAGS)
+src_libfabric_la_LDFLAGS += $(bgq_LDFLAGS)
+src_libfabric_la_LIBADD += $(bgq_LIBS)
+endif
+
+sbin_PROGRAMS = fi_agent0 fi_agent1
+fi_agent1_SOURCES = prov/bgq/src/fi_bgq_agent.c
+fi_agent1_CPPFLAGS = $(AM_CPPFLAGS) $(bgq_CPPFLAGS)
+fi_agent1_LDFLAGS = -all-static $(bgq_LDFLAGS) -Wl,--script=@bgq_driver@/cnk/tools/AppAgent1.lds
+fi_agent0_SOURCES = $(fi_agent1_SOURCES)
```

Comment:
I need the flexability for them to be different executables depending on how I implement advanced optimizations for scalable RMA and triggered ops